File: docs/sdk-examples/aiauto.js
// AIAuto API client — Node.js example (Node 18+, built-in fetch).
// Works the same in React/Next.js API routes — never expose the key in
// browser code; call this from YOUR backend.

const BASE = "https://your-server/api/public/v1";
const KEY = "ak_your_api_key_here";
const HEADERS = { "X-API-Key": KEY, "Content-Type": "application/json" };

async function submitText(prompt) {
  const res = await fetch(`${BASE}/text`, {
    method: "POST", headers: HEADERS, body: JSON.stringify({ prompt }),
  });
  if (!res.ok) throw new Error((await res.json()).detail);
  return res.json();
}

async function submitImage(prompt) {
  const res = await fetch(`${BASE}/images`, {
    method: "POST", headers: HEADERS, body: JSON.stringify({ prompt }),
  });
  if (!res.ok) throw new Error((await res.json()).detail);
  return res.json();
}

async function waitFor(jobId, timeoutMs = 600000) {
  const deadline = Date.now() + timeoutMs;
  while (Date.now() < deadline) {
    const job = await (await fetch(`${BASE}/jobs/${jobId}`,
      { headers: { "X-API-Key": KEY } })).json();
    if (["completed", "failed", "cancelled"].includes(job.status)) return job;
    await new Promise(r => setTimeout(r, 3000));
  }
  throw new Error(`timeout waiting for job ${jobId}`);
}

async function downloadFile(fileId, dest) {
  const fs = await import("node:fs/promises");
  const res = await fetch(`${BASE}/files/${fileId}`, { headers: { "X-API-Key": KEY } });
  await fs.writeFile(dest, Buffer.from(await res.arrayBuffer()));
}

// Express webhook receiver with signature verification
// app.post("/aiauto-hook", express.raw({type: "*/*"}), (req, res) => {
//   const crypto = require("node:crypto");
//   const ts = req.header("X-AIAuto-Timestamp");
//   const sig = req.header("X-AIAuto-Signature");
//   const expected = crypto.createHmac("sha256", WEBHOOK_SECRET)
//     .update(`${ts}.`).update(req.body).digest("hex");
//   if (Math.abs(Date.now()/1000 - Number(ts)) > 300 ||
//       !crypto.timingSafeEqual(Buffer.from(sig), Buffer.from(expected)))
//     return res.status(401).end();
//   const { event, data } = JSON.parse(req.body);
//   console.log(event, data.job_id);
//   res.json({ ok: true });
// });

(async () => {
  let job = await submitText("Write a friendly reminder email about the team meeting");
  console.log("submitted", job.id);
  job = await waitFor(job.id);
  console.log(job.status, job.response ?? job.error);
})();

File: docs/sdk-examples/AiAuto.kt
// AIAuto API client — Android (Kotlin) example using OkHttp + kotlinx.serialization.
// NOTE: never ship an API key inside a distributed app — call your own
// backend, which holds the key, from the app instead.
import kotlinx.coroutines.delay
import okhttp3.MediaType.Companion.toMediaType
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.toRequestBody
import org.json.JSONObject

const val BASE = "https://your-server/api/public/v1"
const val KEY = "ak_your_api_key_here"
val client = OkHttpClient()

fun submitText(prompt: String): JSONObject {
    val body = JSONObject(mapOf("prompt" to prompt)).toString()
        .toRequestBody("application/json".toMediaType())
    val req = Request.Builder().url("$BASE/text")
        .header("X-API-Key", KEY).post(body).build()
    client.newCall(req).execute().use { res ->
        require(res.isSuccessful) { "HTTP ${res.code}" }
        return JSONObject(res.body!!.string())
    }
}

suspend fun waitFor(jobId: String): JSONObject {
    while (true) {
        val req = Request.Builder().url("$BASE/jobs/$jobId")
            .header("X-API-Key", KEY).build()
        client.newCall(req).execute().use { res ->
            val job = JSONObject(res.body!!.string())
            if (job.getString("status") !in listOf("waiting", "processing")) return job
        }
        delay(3000)
    }
}
